File: p6-FutureTechnologies-NetlistReverseEngineering/find_possible_values_test.cpp
#include "code.cpp"

int main() {
    term term1(1, false);
    term term1_negated(1, true);
    term term2(2, false);
    term term2_negated(2, true);
    term term3(3, false);
    term term3_negated(3, true);
    term term4(4, false);
    term term4_negated(4, true);

    std::vector<term> clause1_terms{term1, term2_negated, term4};
    clause clause1(clause1_terms);
    std::vector<term> clause2_terms{term1_negated, term2, term3, term4};
    clause clause2(clause2_terms);
    std::vector<term> clause3_terms{term3_negated, term4};
    clause clause3(clause3_terms);

    std::vector<clause> expression_clauses{clause1, clause2, clause3};
    expression expression1(expression_clauses);

    std::set<bool> possible_values;
    find_possible_values(expression1, possible_values);

    assert(possible_values.size() == 2);

    std::cout << "working ;D" << std::endl;
}

File: p6-FutureTechnologies-NetlistReverseEngineering/find_possible_values_test.py
from sets import Set
from Expression import *
from Algorithms import *

x1 = Term(1, False)
not_x1 = Term(1, True)
x2 = Term(2, False)
not_x2 = Term(2, True)
x3 = Term(3, False)
not_x3 = Term(3, True)
x4 = Term(4, False)
not_x4 = Term(4, True)

clause1 = Clause([x1, not_x2, x4])
clause2 = Clause([not_x1, x2, x3, x4])
clause3 = Clause([not_x3, x4])

expression = Expression([clause1, clause2, clause3])

possible_values = find_possible_values(expression, Set())

assert len(possible_values) == 2
assert True in possible_values
assert False in possible_values

print "working! ;D"
